kmeta/owner_references.go

```diff
@@ -36,3 +37,16 @@ type OwnerRefable interface {
 func NewControllerRef(obj OwnerRefable) *metav1.OwnerReference {
 	return metav1.NewControllerRef(obj.GetObjectMeta(), obj.GetGroupVersionKind())
 }
+
+type ownerRefableDeployment struct {
+	*appsv1.Deployment
+}
+
+func (o *ownerRefableDeployment) GetGroupVersionKind() schema.GroupVersionKind {
+	return appsv1.SchemeGroupVersion.WithKind("Deployment")
+}
```

Comment:
This could just be a type alias instead of a field, but I would like to understand the motivation for this.  Do you have a method that takes `kmeta.OwnerRefable`?  How are you planning to use this?

Reply:
+1 for alias. Yes I'm planning to use it in eventing for mt adapters owned by controllers. 

I can leave it in eventing for now if you prefer and once the PR (yet to be open) gets accepted and merged I can move it here. WDYT?